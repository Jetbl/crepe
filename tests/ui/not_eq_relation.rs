// This test tries to declare an invalid relation struct with a type that does
// not implement the `Hash` or `Eq` traits, and ensures the error message
// points to the relation, rather than a generic span.

use crepe::crepe;

crepe! {
    struct Ok(f64);
}

fn main() {}
