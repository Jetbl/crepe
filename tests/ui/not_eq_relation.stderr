error[E0277]: the trait bound `f64: std::cmp::Eq` is not satisfied
   --> tests/ui/not_eq_relation.rs:8:15
    |
8   |     struct Ok(f64);
    |            -- ^^^ the trait `std::cmp::Eq` is not implemented for `f64`
    |            |
    |            in this derive macro expansion
    |
note: required by a bound in `AssertParamIsEq`
   --> $RUST/core/src/cmp.rs
    |
    | pub struct AssertParamIsEq<T: Eq + ?Sized> {
    |                               ^^ required by this bound in `AssertParamIsEq`
    = note: this error originates in the derive macro `::core::cmp::Eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `f64: Hash` is not satisfied
 --> tests/ui/not_eq_relation.rs:8:15
  |
8 |     struct Ok(f64);
  |            -- ^^^ the trait `Hash` is not implemented for `f64`
  |            |
  |            in this derive macro expansion
  |
  = note: this error originates in the derive macro `::core::hash::Hash` (in Nightly builds, run with -Z macro-backtrace for more info)
